### PR TITLE
fix: detect source paths from compile_commands.json by default

### PR DESCRIPTION
**Pull request by antipeon**

### CODE REVIEW

**Comment by antipeon** [comment]
Nothing in srcFolders -> auto generate from compile_commands.json
Screenshot (4)
Screenshot (6)

**Comment by antipeon** [comment]
something in srcFolders -> use it for generateForProject
Screenshot (9)
Screenshot (10)

File: server/src/Server.cpp [hunk 1]
```diff
@@ -256,7 +256,7 @@ Status Server::TestsGenServiceImpl::ProcessBaseTestRequest(BaseTestGen &testGen,
         }
         auto generator = std::make_shared<KleeGenerator>(
             testGen.projectContext, testGen.settingsContext,
-            testGen.serverBuildDir, testGen.sourcePaths, testGen.compilationDatabase, typesHandler,
+            testGen.serverBuildDir, testGen.testingMethodsSourcePaths, testGen.compilationDatabase, typesHandler,
```
Comment (ladisgin) [inline server/src/Server.cpp:259]: rename back

File: server/src/Server.h [hunk 2]
```diff
@@ -119,7 +119,7 @@ class Server {
 
                 MEASURE_FUNCTION_EXECUTION_TIME
 
-                TestGenT testGen{ request, testsWriter.get(), testMode };
+                TestGenT testGen{ request, testsWriter.get(), testMode};
```
Comment (ladisgin) [inline server/src/Server.h:122]: return space

File: server/src/testgens/ProjectTestGen.cpp [hunk 3]
```diff
@@ -22,22 +23,16 @@ ProjectTestGen::ProjectTestGen(const testsgen::ProjectRequest &request,
...
-        Paths::filterPathsByDirNames(sourcePathsCandidates, requestSourcePaths, { ".c", ".cpp", ".cc" });
+    // sourcePathsCandidates are from compile_commands.json
+    vector<fs::path> sourcePathsCandidates = getSourcePathCandidates();
+    // requestSourcePaths are from settings.json
+    auto requestSourcePaths = getRequestSourcePaths();
+    sourcePaths = sourcePathsCandidates;
     testingMethodsSourcePaths = sourcePaths;
     setInitializedTestsMap();
+    if (autoDetect) {
+        autoDetectSourcePathsIfNotEmpty();
```
Comment (ladisgin) [inline server/src/testgens/ProjectTestGen.cpp:34]: add if else

File: server/src/testgens/ProjectTestGen.h [hunk 4]
```diff
@@ -26,6 +27,12 @@ class ProjectTestGen : public BaseTestGen {
 
 private:
     testsgen::ProjectRequest const *const request;
+
+    vector<fs::path> getSourcePathCandidates();
```
Comment (ladisgin) [inline server/src/testgens/ProjectTestGen.h:31]: maybe make const

**Review by ladisgin — CHANGES_REQUESTED** [review]

File: server/src/testgens/ProjectTestGen.cpp [hunk 5]
```diff
@@ -11,7 +11,8 @@
 
 ProjectTestGen::ProjectTestGen(const testsgen::ProjectRequest &request,
                                ProgressWriter *progressWriter,
-                               bool testMode)
+                               bool testMode,
+                               bool isGenerateForProject)
```
Comment (ladisgin) [inline server/src/testgens/ProjectTestGen.cpp:15]: change name back

File: server/src/Paths.cpp [hunk 6]
```diff
@@ -355,4 +354,8 @@ namespace Paths {
 
     //endregion
 
+    const std::vector<std::string> CXXSourceExtensions({".cc", ".cp", ".cpp", ".c++", ".cxx"});
+    const std::vector<std::string> headerExtensions({".hh", ".hpp", ".hxx"});
```
Comment (ladisgin) [inline server/src/Paths.cpp:358]: rename

File: server/src/Paths.cpp [hunk 7]
```diff
@@ -355,4 +354,8 @@ namespace Paths {
...
+    const std::string CSourceExtension = ".c";
```
Comment (ladisgin) [inline server/src/Paths.cpp:359]: change to vector

**Review by ladisgin — APPROVED** [review]

**Pull request merged**